### ISSUE TITLE
test_original_air_date_should_be_a_date fails in clean chroot

### ISSUE THREAD

**Issue by dvzrv** [alberanid/imdbpy#135]
#### Issue description
Running the tests in a clean chroot during package build (for Arch Linux), I get one failed test:

```
============================= test session starts ==============================
platform linux -- Python 3.6.4, pytest-3.4.2, py-1.5.2, pluggy-0.6.0
rootdir: /build/python-imdbpy/src/python-imdbpy-6.3, inifile:
plugins: cov-2.5.1
collected 158 items

tests/test_http_character_main.py s                                      [  0%]
tests/test_http_chart_bottom.py ...                                      [  2%]
tests/test_http_chart_top.py .....                                       [  5%]
tests/test_http_company_main.py .                                        [  6%]
tests/test_http_movie_combined.py ..............ss..............ss...... [ 30%]
............F.......ss............................                       [ 62%]
tests/test_http_movie_keywords.py ..                                     [ 63%]
tests/test_http_movie_plot.py ...                                        [ 65%]
tests/test_http_movie_reviews.py .                                       [ 65%]
tests/test_http_movie_sites.py .                                         [ 66%]
tests/test_http_movie_taglines.py ...                                    [ 68%]
tests/test_http_movie_votes.py ......                                    [ 72%]
tests/test_http_person_bio.py ..................                         [ 83%]
tests/test_http_person_main.py ......                                    [ 87%]
tests/test_http_person_otherworks.py ...                                 [ 89%]
tests/test_http_search_character.py s                                    [ 89%]
tests/test_http_search_company.py .                                      [ 90%]
tests/test_http_search_keyword.py ....                                   [ 93%]
tests/test_http_search_movie.py .......                                  [ 97%]
tests/test_http_search_movie_keyword.py ...                              [ 99%]
tests/test_http_search_person.py .                                       [100%]

=================================== FAILURES ===================================
___________________ test_original_air_date_should_be_a_date ____________________

movie_combined_details = <function movie_combined_details.<locals>.retrieve at 0x61d05d3a7d90>

    def test_original_air_date_should_be_a_date(movie_combined_details):
        page = movie_combined_details('dr who ep blink')
        data = parser.parse(page)['data']
>       assert data['original air date'] == '09 Jun 2007'
E       AssertionError: assert '17 Oct 2012' == '09 Jun 2007'
E         - 17 Oct 2012
E         + 09 Jun 2007

tests/test_http_movie_combined.py:369: AssertionError
=============== 1 failed, 149 passed, 8 skipped in 83.18 seconds ===============

```

#### Version of IMDbPY, Python and OS

- **Python:** Python 3.6.4
- **IMDbPY:** 6.3
- **OS:** `uname_result(system='Linux', node='dvzrv', release='4.15.7-1-hardened', version='#1 SMP PREEMPT Wed Feb 28 20:56:46 CET 2018', machine='x86_64', processor='')
`

#### Steps to reproduce the issue

```
python setup.py build

export export PYTHONPATH=${srcdir}/${pkgname}-${pkgver}/build:${PYTHONPATH} #only relevant in the build environment
py.test

```

#### What's the expected result?

- tests should pass

#### What's the actual result?

- `test_original_air_date_should_be_a_date` fails

#### Additional details

I guess the date has changed. I'm disabling this test.

**Comment by dvzrv** [comment]
Whoops, that should've read 6.3 as the imdbpy version ;-)

**Comment by alberanid** [comment]
Hi @dvzrv, thanks for the report.  Closing it for the moment, since I'm unable to reproduce, and can't find the dte you got in the page sources.  It's possible that it was a temporary glitch on their side.

Thanks!

**Comment by dvzrv** [comment]
@alberanid well, why do you close it then? This obviously is an issue (to me and other packagers for this software). Just because you can't reproduce this, doesn't mean others can't or shouldn't be able to read about it (when they package this for some other distro).

I just tested and can reproduce this (also in a local - non-chroot - environment):

```
=================================== FAILURES ===================================
___________________ test_original_air_date_should_be_a_date ____________________              
                                                                                
movie_combined_details = <function movie_combined_details.<locals>.retrieve at 0x631ec0d7ce18>
                                                                                
    def test_original_air_date_should_be_a_date(movie_combined_details):
        page = movie_combined_details('dr who ep blink')                        
        data = parser.parse(page)['data']                             
>       assert data['original air date'] == '09 Jun 2007'                      
E       AssertionError: assert '17 Oct 2012' == '09 Jun 2007'         
E         - 17 Oct 2012                                                   
E         + 09 Jun 2007                                                  
                                                                                
tests/test_http_movie_combined.py:369: AssertionError                    
______ test_found_many_result_should_contain_correct_number_of_companies _______          
                                                                       
search_company = <function search_company.<locals>.retrieve at 0x631ec0de3378>  
                                                                                          
    @mark.fragile                                                               
    def test_found_many_result_should_contain_correct_number_of_companies(search_company):
        page = search_company('pixar')                                          
        data = parser.parse(page)['data']                           
>       assert len(data) >= 38                                                                  
E       assert 0 >= 38                                                      
E        +  where 0 = len([])                                                                   
                                                                                                                                           
tests/test_http_search_company.py:22: AssertionError                                            
____________ test_found_one_result_should_be_list_with_one_keyword _____________                                                           
                                                                               
search_keyword = <function search_keyword.<locals>.retrieve at 0x631ec0ed19d8>  
                                                                                
    @mark.fragile                                                               
    def test_found_one_result_should_be_list_with_one_keyword(search_keyword): 
        page = search_keyword('zoolander')                                     
        data = parser.parse(page)['data']                                            
>       assert data == ['colander']                                            
E       AssertionError: assert [] == ['colander']                                    
E         Right contains more items, first extra item: 'colander'              
E         Use -v to get the full diff                                          
                                                                                
tests/test_http_search_keyword.py:22: AssertionError                           
_______ test_found_many_result_should_contain_correct_number_of_keywords _______
                                                                                
search_keyword = <function search_keyword.<locals>.retrieve at 0x631ec0ed19d8>  
                                                                                         
    @mark.fragile                                                                         
    def test_found_many_result_should_contain_correct_number_of_keywords(search_keyword):
        page = search_keyword('messiah')                                       
        data = parser.parse(page)['data']                                                    
>       assert 40 < len(data) < 50                                              
E       assert 40 < 0                                                           
E        +  where 0 = len([])                                                                 
                                                                                
tests/test_http_search_keyword.py:29: AssertionError                                          
____________ test_found_too_many_result_should_contain_200_keywords ____________
                                                                        
search_keyword = <function search_keyword.<locals>.retrieve at 0x631ec0ed19d8>  
                                                                      
    def test_found_too_many_result_should_contain_200_keywords(search_keyword):           
        page = search_keyword('computer')                          
        data = parser.parse(page)['data']                                         
>       assert len(data) == 200                                          
E       assert 0 == 200                                                                   
E        +  where 0 = len([])                                         
                                                                                          
tests/test_http_search_keyword.py:35: AssertionError                   
_____________ test_found_one_result_should_be_list_with_one_movie ______________
                                                                                          
search_movie = <function search_movie.<locals>.retrieve at 0x631ec0ed1d08>      
                                                                                          
    def test_found_one_result_should_be_list_with_one_movie(search_movie):                
        page = search_movie('od instituta do proizvodnje')          
        data = parser.parse(page)['data']                                                       
>       assert data == [                       
            ('0483758', {'kind': 'short', 'title': 'Od instituta do proizvodnje', 'year': 1971})
        ]                                                                                                                                  
E       AssertionError: assert [] == [('0483758', {'kind': 'short...roizvodnje', 'year': 1971})]
E         Right contains more items, first extra item: ('0483758', {'kind': 'short', 'title': 'Od instituta do proizvodnje', 'year': 1971})
E         Use -v to get the full diff                                           
                                                                                
tests/test_http_search_movie.py:21: AssertionError                              
________ test_found_many_result_should_contain_correct_number_of_movies ________
                                                                               
search_movie = <function search_movie.<locals>.retrieve at 0x631ec0ed1d08>     
                                                                                     
    @mark.fragile                                                                     
    def test_found_many_result_should_contain_correct_number_of_movies(search_movie):
        page = search_movie('ace in the hole')                                 
        data = parser.parse(page)['data']                                      
>       assert 185 < len(data) < 200                                            
E       assert 185 < 0                                                         
E        +  where 0 = len([])                                                   
                                                                                
tests/test_http_search_movie.py:30: AssertionError                                        
____________ test_found_too_many_result_should_contain_200_keywords ____________
                                                                        
search_keyword = <function search_keyword.<locals>.retrieve at 0x631ec0ed19d8>  
                                                                      
    def test_found_too_many_result_should_contain_200_keywords(search_keyword):
        page = search_keyword('computer')                             
        data = parser.parse(page)['data']                                 
>       assert len(data) == 200                                          
E       assert 0 == 200                                                         
E        +  where 0 = len([])                                            
                                                                                          
tests/test_http_search_keyword.py:35: AssertionError                   
_____________ test_found_one_result_should_be_list_with_one_movie ______________
                                                                                          
search_movie = <function search_movie.<locals>.retrieve at 0x631ec0ed1d08>      
                                                                                          
    def test_found_one_result_should_be_list_with_one_movie(search_movie):      
        page = search_movie('od instituta do proizvodnje')          
        data = parser.parse(page)['data']                                                       
>       assert data == [                                                    
            ('0483758', {'kind': 'short', 'title': 'Od instituta do proizvodnje', 'year': 1971})
        ]                                                                                                                                  
E       AssertionError: assert [] == [('0483758', {'kind': 'short...roizvodnje', 'year': 1971})]
E         Right contains more items, first extra item: ('0483758', {'kind': 'short', 'title': 'Od instituta do proizvodnje', 'year': 1971})
E         Use -v to get the full diff                                          
                                                                                
tests/test_http_search_movie.py:21: AssertionError                              
________ test_found_many_result_should_contain_correct_number_of_movies ________
                                                                               
search_movie = <function search_movie.<locals>.retrieve at 0x631ec0ed1d08>     
                                                                                     
    @mark.fragile                                                              
    def test_found_many_result_should_contain_correct_number_of_movies(search_movie):
        page = search_movie('ace in the hole')                                 
        data = parser.parse(page)['data']                                      
>       assert 185 < len(data) < 200                                            
E       assert 185 < 0                                                         
E        +  where 0 = len([])                                                   
                                                                                
tests/test_http_search_movie.py:30: AssertionError                              
_____________ test_found_too_many_result_should_contain_200_movies _____________         
                                                                                          
search_movie = <function search_movie.<locals>.retrieve at 0x631ec0ed1d08>               
                                                                               
    def test_found_too_many_result_should_contain_200_movies(search_movie):                  
        page = search_movie('matrix')                                           
        data = parser.parse(page)['data']                                       
>       assert len(data) == 200                                                               
E       assert 0 == 200                                                         
E        +  where 0 = len([])                                                                 
                                                                                
tests/test_http_search_movie.py:36: AssertionError                      
_____________ test_found_many_result_should_contain_correct_movie ______________
                                                                      
search_movie = <function search_movie.<locals>.retrieve at 0x631ec0ed1d08>                
                                                                   
    def test_found_many_result_should_contain_correct_movie(search_movie):        
        page = search_movie('matrix')                                    
        data = parser.parse(page)['data']                                                 
        movies = dict(data)                                           
>       assert movies['0133093'] == {'title': 'The Matrix', 'kind': 'movie', 'year': 1999}
E       KeyError: '0133093'                                            
                                                                                
tests/test_http_search_movie.py:43: KeyError                                              
______________________ test_found_movie_should_have_kind _______________________
                                                                                          
search_movie = <function search_movie.<locals>.retrieve at 0x631ec0ed1d08>                
                                                                    
    def test_found_movie_should_have_kind(search_movie):                                        
        page = search_movie('matrix')          
        data = parser.parse(page)['data']                                                       
        movies = dict(data)                                                                                                                
>       assert movies['0106062'] == {'title': 'Matrix', 'kind': 'tv series', 'year': 1993}      
E       KeyError: '0106062'                                                                                                                
                                                                                
tests/test_http_search_movie.py:50: KeyError                                    
___________________ test_found_movie_should_have_imdb_index ____________________
                                                                                
search_movie = <function search_movie.<locals>.retrieve at 0x631ec0ed1d08>     
                                                                               
    def test_found_movie_should_have_imdb_index(search_movie):                       
        page = search_movie('blink')                                                  
        data = parser.parse(page)['data']                                            
        movies = dict(data)                                                    
>       assert movies['4790262'] == {'title': 'Blink', 'imdbIndex': 'IV',      
                                     'kind': 'movie', 'year': 2015}             
E       KeyError: '4790262'                                                    
                                                                                
tests/test_http_search_movie.py:57: KeyError                                    
________ test_found_many_result_should_contain_correct_number_of_movies ________
                                                                                         
search_movie_keyword = <function search_movie_keyword.<locals>.retrieve at 0x631ec0ed1730>
                                                                                         
    @mark.fragile                                                              
    def test_found_many_result_should_contain_correct_number_of_movies(search_movie_keyword):
        page = search_movie_keyword('colander')                                 
        data = parser.parse(page)['data']                                       
>       assert len(data) == 4                                                                 
E       assert 0 == 4                                                           
E        +  where 0 = len([])                                                                 
                                                                                
tests/test_http_search_movie_keyword.py:22: AssertionError              
_____________ test_found_too_many_result_should_contain_50_movies ______________
                                                                      
search_movie_keyword = <function search_movie_keyword.<locals>.retrieve at 0x631ec0ed1730>
                                                                   
    def test_found_too_many_result_should_contain_50_movies(search_movie_keyword):
        page = search_movie_keyword('computer')                          
        data = parser.parse(page)['data']                                                 
>       assert len(data) == 50                                        
E       assert 0 == 50                                                                    
E        +  where 0 = len([])                                          
                                                                                
tests/test_http_search_movie_keyword.py:28: AssertionError                                
_____________ test_found_many_result_should_contain_correct_movie ______________
                                                                                          
search_movie_keyword = <function search_movie_keyword.<locals>.retrieve at 0x631ec0ed1730>
                                                                    
    def test_found_many_result_should_contain_correct_movie(search_movie_keyword):              
        page = search_movie_keyword('colander')
        data = parser.parse(page)['data']                                                       
        movies = dict(data)                                                                                                                
>       assert movies['0382932'] == {'title': 'Ratatouille', 'kind': 'movie', 'year': 2007}     
E       KeyError: '0382932'                                                                                                                
                                                                                
tests/test_http_search_movie_keyword.py:35: KeyError                            
________ test_found_many_result_should_contain_correct_number_of_people ________
                                                                                
search_person = <function search_person.<locals>.retrieve at 0x631ec0ed1f28>   
                                                                               
    @mark.fragile                                                                    
    def test_found_many_result_should_contain_correct_number_of_people(search_person):
        page = search_person('keanu reeves')                                         
        data = parser.parse(page)['data']                                      
>       assert len(data) == 4                                                  
E       assert 0 == 4                                                           
E        +  where 0 = len([])                                                  
                                                                                
tests/test_http_search_person.py:22: AssertionError                             
============== 15 failed, 135 passed, 8 skipped in 84.32 seconds =============== 
```       

Now with even more tests failing.

Is this something that has happened before, or is to be expected to happen often (when working with IMDB)?
I'm asking, because we're implementing reproducible builds for Arch Linux and for that to work, it doesn't makes sense to have the tests run, if they're failing all the time anyways.

**Comment by alberanid** [comment]
the first test fails because IMDb seems to serve different data depending on the source IP.  So yes, it may happens often.  I've fixed it changing the test to use a regular expression to identify a date.

The other failures: I'm able to reproduce them, starting from a few hours ago.  Yes, seems like IMDb changed the format of the html pages returned doing a search.  Also this happens quite often; I'll open a separate issue for that, since it's more serious.

Thanks.

**Comment by alberanid** [comment]
#136 fixed in master; we'll probably have a 6.4 release soon.

**Comment by alberanid** [comment]
version 6.4 released.  Closing this issue since all tests should now pass.  If that's not the case, please open it again (or another issue).

Thanks!
